src/js/content.js

```diff
@@ -22,6 +26,28 @@ window.addEventListener('message', (event) => {
   }
 })
 
+function getExtensionStatus() {
```

Comment:
There are a number of functional updates going on with this change, would you be willing to add some tests before shipping?